### DISCUSSION TITLE
Open BackstageTab programmatically

### DISCUSSION THREAD

**Discussion by Taldrit78** [fluentribbon/Fluent.Ribbon#1189] [Q&A] [answered]
We have created a RibbonService and inside we defined two properties of the kind "BackstageTabItem".
One is the HomeTab and one is the HelpTab.
Now we want to show the HelpTab on press of F1.
But if we set the IsSelected on these BackstageTabItem's nothing happens.
Is there any way to show these tabs programmatically?
p.s.: Showing them by mouse click works fine.

Ok, maybe you need some more infos on how we integrated it.
This is the main code:

            BackstageTabControl backstage = new BackstageTabControl();
            // Then we fill the BackstageTabControl
            System.Windows.Controls.ScrollViewer scrollViewer = new System.Windows.Controls.ScrollViewer()
            {
                Content = backstage,
                HorizontalScrollBarVisibility = System.Windows.Controls.ScrollBarVisibility.Auto,
                VerticalScrollBarVisibility = System.Windows.Controls.ScrollBarVisibility.Auto
            };
            Ribbon.Menu = new Backstage() { Header = mRibbonData[0].Header, MinWidth = 100, Content = scrollViewer, KeyTip = mRibbonData[0].KeyTip };

But I cannot even open the Menu programmatically... what do I miss here?

**Comment by batzen** [comment, marked as answer]
I don't know how you are trying to open the backstage and select an item.

Just added code that shows how it works to the showcase application.
The commit is https://github.com/fluentribbon/Fluent.Ribbon/commit/eeadf49f093faaa368e635fe70bb2b81e9e49adc

Why are you putting the `BackstageTabControl` inside a `ScrollViewer`, by the way?

**Comment by Taldrit78** [comment]
Thank you. And I don't know why he (my colleague) did it. He doesn't remember either. We guess, that it is, because, even if the helppage inside is not really big, the user can reduce the main window of our application to such a small size that it is required.

Oh and we use this code now in the OnPreviewKeyDown event handler:

            if (e.Key == Key.F1
                && Ribbon.Menu is Fluent.Backstage backstage
                && backstage.Content is ScrollViewer scrollViewer
                && scrollViewer.Content is Fluent.BackstageTabControl tabControl)
            {
                backstage.IsOpen = true;
                Fluent.BackstageTabItem item = tabControl.Items.OfType<Fluent.BackstageTabItem>().FirstOrDefault(x => x.Name == "Helptab");
                if (item != null)
                {
                    item.IsSelected = true;
                }
            }
